packages/firestore/src/local/remote_document_change_buffer.ts

```diff
@@ -25,6 +25,16 @@ import { PersistenceTransaction } from './persistence';
 import { PersistencePromise } from './persistence_promise';
 import { SnapshotVersion } from '../core/snapshot_version';
 
+/**
+ * Represents a document change to be applied to remote document cache.
+ */
+interface RemoteDocumentChange {
+  // The document in this change, null if it is a removal from the cache.
+  readonly maybeDocument: MaybeDocument | null;
+  // The timestamp when this change is read.
+  readonly readTime?: SnapshotVersion;
```

Comment:
The use of "null" and "undefined" for "not set" is a bit inconsistent here. 

FWIW, we could drop readTime if we add it to MaybeDocument.